wordpress/wp-content/plugins/memberful-wp/src/user/map.php

```diff
@@ -146,7 +146,11 @@ public function ensure_present() {
     $user_id = $this->user_exists ? $this->update_user() : $this->create_user();
 
     if ( is_wp_error( $user_id ) ) {
-      $user_id->add_data( array_merge( (array) $user_id->get_error_data(), compact('user_data') ) );
+      if (isset($user_data)) {
+        $user_id->add_data( array_merge( (array) $user_id->get_error_data(), compact('user_data') ) );
+      } else {
+        $user_id->add_data( array_merge( (array) $user_id->get_error_data(), array() ) );
```

Comment:
I presume that this should work, too:

```suggestion
        $user_id->add_data( (array) $user_id->get_error_data() );
```

Reply:
Thanks for pointing out the `user_data` variable! I was curious why I couldn't find it when searching, and I (incorrectly) assumed it was a WP or plugin method somewhere.

I've also updated the code to remove the `compact` and `array_merge`. It really didn't make sense to merge an empty array in the first place, not sure what I was thinking there 😅.